agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/localstorage/LocalFileSender.java

```diff
@@ -60,7 +60,9 @@ public void run() {
     try {
       LocalFileLoader.PersistedFile persistedFile = localFileLoader.loadTelemetriesFromDisk();
       if (persistedFile != null) {
-        CompletableResultCode resultCode = telemetryChannel.sendRawBytes(persistedFile.rawBytes);
+        assert (persistedFile.instrumentationKey != null);
```

Comment:
asserts aren't typically enabled, so rarely used in Java

one option:

```suggestion
        if (persistedFile.instrumentationKey == null) {
          throw new IllegalStateException("...")
        }
```

but better option here is to validate in PersistedFile constructor and throw IllegalArgumentException